src/layers/core/icon-layer/icon-layer.js

```diff
@@ -107,7 +107,14 @@ export default class IconLayer extends Layer {
   updateState({oldProps, props, changeFlags}) {
     super.updateState({props, oldProps, changeFlags});
 
-    const {iconAtlas} = props;
+    const {iconAtlas, iconMapping} = props;
+
+    if (oldProps.iconMapping !== iconMapping) {
```

Comment:
As far as I can tell, the listed attributes depend only on the `iconMapping` (in the sense that the functions calculating those attributes refer to `iconMapping`) but they do not depend on the `iconAtlas`. So recalculating them if only `iconAtlas` changes should be a noop - i.e generate identical results.

Agree it is better to regenerate the attributes too often (small perf hit) rather than too seldom (incorrect rendering)...